### PR TITLE
Supported platforms revisited

### LINKED ISSUE #276: Review supported-platform page
This is a followup issue to #273. For context, see https://github.com/etcd-io/website/pull/273#issuecomment-837366577.

### PR DESCRIPTION
**Pull request by chalin**
Closes #276

Preview: https://deploy-preview-277--etcd.netlify.app/docs/next/op-guide/supported-platform/

/reviewer @gyuho 
/cc @spzala @glevand @nate-double-u

### CODE REVIEW

**Review by gyuho — APPROVED** [review]

File: content/en/docs/next/op-guide/supported-platform.md [hunk 1]
```diff
@@ -1,44 +1,71 @@
...
+platform's support tier:
+
+- **Tier-1** platforms are _fully supported_ by [etcd maintainers][], and etcd
+  is guaranteed to pass all tests including functional and stress tests.
+- On **tier-2** platforms, etcd is guaranteed to pass integration and end-to-end
+  tests but not necessarily functional or stress tests.
+- While etcd may build and run on **tier-3** platforms and unlisted
+  architectures, it may or may not pass any tests, and so it is considered
+  _unstable_ &mdash; caveat emptor.
+
```
[the remark this answers is not in the document]
Comment (chalin) [reply content/en/docs/next/op-guide/supported-platform.md:19]: Regarding the "unlisted" bullet: tier-3 guarantees that etcd will at least build on a platform (see the "Supporting a new platform" section) -- that's too strong a claim to make for all unlisted platforms IMHO.

Regarding "tier-3" being "unsupported": the tiers are called "support tiers", so it seems inconsistent to me to say that platforms in tier-3 are unsupported.

We could address the points mentioned above by stating that unlisted platforms are unsupported, which to me is implicitly understood -- anything not mentioned under "Current support" is unsupported.

Thoughts?

[the remark this answers is not in the document]
Comment (chalin) [reply content/en/docs/next/op-guide/supported-platform.md:19]: Come to think of it, what I wrote in the previous comment is an argument for remove the "and unlisted architectures" from the text.

File: content/en/docs/next/op-guide/supported-platform.md [hunk 2]
```diff
@@ -1,44 +1,71 @@
...
-title: Supported systems
+title: Supported platforms
 weight: 4800
-description: etcd support status for common architectures & operating systems
+description: etcd support for common architectures & operating systems
 ---
 
+## Support tiers
+
+Etcd runs on different platforms, but the guarantees it provides depends on a
```
[the remark this answers is not in the document]
Comment (chalin) [reply content/en/docs/next/op-guide/supported-platform.md:9]: Fixed

File: content/en/docs/next/op-guide/supported-platform.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chalin) [reply content/en/docs/next/op-guide/supported-platform.md:18]: Thanks for the feedback @glevand. PTAL at the edits I've made.

[the remark this answers is not in the document]
Comment (chalin) [reply content/en/docs/next/op-guide/supported-platform.md:18]: Oh, our comments crossed :).

@glevand: let me know if you're ok with the new wording or would prefer it further abridged along the lines of your last suggestions.

**Pull request merged**